sirepo/template/controls.py

```diff
@@ -165,17 +166,29 @@ def write_parameters(data, run_dir, is_parallel):
 
 
 def _add_monitor(data):
-    if list(filter(lambda el: el.type == 'MONITOR', data.models.elements)):
+    o = list(filter(lambda el: el.type == 'MONITOR', data.models.elements))
+    a = list(filter(lambda el: el.type == 'MARKER', data.models.elements))
+    if o and not a:
         return
-    m = PKDict(
-        _id=LatticeUtil.max_id(data) + 1,
-        name='M_1',
-        type='MONITOR',
-    )
-    data.models.elements.append(m)
-    assert len(data.models.beamlines) == 1, \
-        f'expecting 1 beamline={data.models.beamlines}'
-    data.models.beamlines[0]['items'].append(m._id)
+    if not o and not a:
+        m = PKDict(
+            _id=LatticeUtil.max_id(data) + 1,
+            name='M1',
+            type='MONITOR',
+        )
+        data.models.elements.append(m)
+        assert len(data.models.beamlines) == 1, \
+            f'expecting 1 beamline={data.models.beamlines}'
+        data.models.beamlines[0]['items'].append(m._id)
+        return
+    for i, e in enumerate(data.models.elements):
```

Comment:
I think you could do this part as the first step in the method and then you don't need to count markers, only monitors.